webv2/updateTableSchema/removeColumn.go

```diff
@@ -0,0 +1,331 @@
+)
+
+// removeColumn remove given column from schema.
+func removeColumn(table string, colName string, Conv *internal.Conv) {
+
+	sp := Conv.SpSchema[table]
+
+	removeColumnFromCurrentTableSchema(Conv, sp, table, colName)
+
+	// update foreignKey relationship Table column names
+	for i, _ := range sp.Fks {
+
+		removeColumnFromForeignkeyTableSchema(Conv, sp, i, colName)
+
+	}
+
+	for _, sp := range Conv.SpSchema {
+
+		for j := 0; j < len(sp.Fks); j++ {
+			if sp.Fks[j].ReferTable == table {
+				removeColumnFromForeignkeyReferTableSchema(Conv, sp, sp.Name, colName)
+			}
+
+		}
+
+	}
+
+	isParent, parentSchemaTable := IsParent(table)
+
+	if isParent {
+
+		removeColumnFromParentTableSchema(Conv, parentSchemaTable, colName)
+
+	}
+
+	childSchemaTable := Conv.SpSchema[table].Parent
+
+	if childSchemaTable != "" {
+
+		removeColumnFromChildTableSchema(Conv, parentSchemaTable, colName)
+	}
+}
+
+// removeColumnFromCurrentTableSchema remove given column from current table schema.
+func removeColumnFromCurrentTableSchema(Conv *internal.Conv, sp ddl.CreateTable, table string, colName string) {
+	// step I
+	sp = removeColumnFromSpannerColDefs(sp, colName)
+
+	// step II
+	sp = removeColumnFromSpannerPK(sp, colName)
+
+	// step III
+	sp = removeColumnFromSpannerSecondaryIndex(sp, colName)
+
+	// step IV
+	sp = removeColumnFromSpannerForeignkeyColumns(sp, colName)
+
+	// step V
+	sp = removeColumnFromSpannerForeignkeyReferColumns(sp, colName)
+
+	// step VI
+	sp = removeColumnFromSpannerColNames(sp, colName)
+
+	// step VII
+	removeSpannerSchemaIssue(table, colName, Conv)
+
+	// step VIII
+	removeColumnFromToSpannerToSource(table, colName, Conv)
+
+	Conv.SpSchema[table] = sp
+
+}
+
+// removeColumnFromForeignkeyTableSchema remove given column from Foreignkey Table relationship Schema.
+func removeColumnFromForeignkeyTableSchema(Conv *internal.Conv, sp ddl.CreateTable, index int, colName string) {
+
+	relationTable := sp.Fks[index].ReferTable
+	relationTableSp := Conv.SpSchema[relationTable]
+
+	relationTableSp = removeColumnFromSpannerColNames(relationTableSp, colName)
+	relationTableSp = removeColumnFromSpannerColDefs(relationTableSp, colName)
+	relationTableSp = removeColumnFromSpannerPK(relationTableSp, colName)
+	relationTableSp = removeColumnFromSpannerSecondaryIndex(relationTableSp, colName)
+	relationTableSp = removeColumnFromSpannerForeignkeyColumns(relationTableSp, colName)
+	relationTableSp = removeColumnFromSpannerForeignkeyReferColumns(relationTableSp, colName)
+
+	removeSpannerSchemaIssue(relationTable, colName, Conv)
+	removeColumnFromToSpannerToSource(relationTable, colName, Conv)
+
+	Conv.SpSchema[relationTable] = relationTableSp
+}
+
+// removeColumnFromForeignkeyReferTableSchema remove given column from Foreign key Refer Table Schema.
+func removeColumnFromForeignkeyReferTableSchema(Conv *internal.Conv, referTable ddl.CreateTable, table string, colName string) {
```

Comment:
No need of this function , reuse - removeColumnFromCurrentTableSchema(). Rename the method to solve both the purposes.

Reply:
Updated